### PR TITLE
Complement ProcessRole and ProcessClass

### PR DESCRIPTION
**Pull request by mengzhbj**
# Description

ProcessRole and ProcessClass, the two types didn't include all the existing roles and classes, this PR complements them.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Discussion

No.

## Testing

No.

## Documentation

No, didn't need to update any doc.

## Follow-up

No.

### CODE REVIEW

File: api/v1beta2/foundationdb_status.go [hunk 1]
```diff
@@ -296,10 +296,31 @@ type FoundationDBStatusBackupTag struct {
...
+	// ProcessRoleProxy model for FDB proxy role
+	ProcessRoleProxy ProcessRole = "proxy"
+	// ProcessRoleGrvProxy model for FDB grv_proxy role
+	ProcessRoleGrvProxy ProcessRole = "grv_proxy"
+	// ProcessRoleCommitProxy model for FDB commit_proxy role
+	ProcessRoleCommitProxy ProcessRole = "commit_proxy"
+	// ProcessRoleResolver model for FDB resolver role
+	ProcessRoleResolver ProcessRole = "resolver"
+	// ProcessRoleData_Distributor model for FDB data_distributor role
+	ProcessRoleData_Distributor ProcessRole = "data_distributor"
```
Comment (johscheuer) [inline api/v1beta2/foundationdb_status.go:321]:
```suggestion
	ProcessRoleDataDistributor ProcessRole = "data_distributor"
```

Go doesn't use `_` in it's name.

**Review by johscheuer — CHANGES_REQUESTED** [review]
Thanks for the PR, once the lint check passes we can merge this one 👍

File: api/v1beta2/foundationdb_status.go (hunk 1 above)
Comment (mengzhao-ibm) [reply api/v1beta2/foundationdb_status.go:321]: Ah, I was a little careless... Will be more careful next time. Thanks!

**Comment by mengzhbj** [comment]
Hi @johscheuer, the PR Builder check failed and I found following errors in logs/test_operator_upgrades.log:
``` [FAIL] Operator Upgrades upgrading a cluster with a partitioned pod [It] Upgrade from 6.3.25 to 7.1.27 with partitioned Pod
  /codebuild/output/src833382869/src/github.com/FoundationDB/fdb-kubernetes-operator/fdb-kubernetes-tests/tests/fixtures/status.go:35
  [FAIL] Operator Upgrades upgrading a cluster with a partitioned pod which eventually gets replaced [It] Upgrade from 6.3.25 to 7.1.27 with partitioned Pod which eventually gets replaced
  /codebuild/output/src833382869/src/github.com/FoundationDB/fdb-kubernetes-operator/fdb-kubernetes-tests/tests/fixtures/status.go:35
  [FAIL] Operator Upgrades upgrading a cluster with a crash looping sidecar process [It] Upgrade from 6.3.25 to 7.1.27 with crash-looping sidecar
  /codebuild/output/src833382869/src/github.com/FoundationDB/fdb-kubernetes-operator/fdb-kubernetes-tests/tests/fixtures/status.go:35
  [FAIL] Operator Upgrades upgrading a cluster and one coordinator is not restarted [It] Upgrade from 6.3.25 to 7.1.27 with one coordinator not being restarted
  /codebuild/output/src833382869/src/github.com/FoundationDB/fdb-kubernetes-operator/fdb-kubernetes-tests/tests/test_operator_upgrades/operator_upgrades_test.go:443
  ```
 Can you explain a bit how to find fdb-kubernetes-tests? I've tried https://github.com/FoundationDB/fdb-kubernetes-tests but no luck.

**Review by johscheuer — APPROVED** [review]
> Hi @johscheuer, the PR Builder check failed and I found following errors in logs/test_operator_upgrades.log:
>
 > [FAIL] Operator Upgrades upgrading a cluster with a partitioned pod [It] Upgrade from 6.3.25 to 7.1.27 with partitioned Pod
  /codebuild/output/src833382869/src/github.com/FoundationDB/fdb-kubernetes-operator/fdb-kubernetes-tests/tests/fixtures/status.go:35
  [FAIL] Operator Upgrades upgrading a cluster with a partitioned pod which eventually gets replaced [It] Upgrade from 6.3.25 to 7.1.27 with partitioned Pod which eventually gets replaced
  /codebuild/output/src833382869/src/github.com/FoundationDB/fdb-kubernetes-operator/fdb-kubernetes-tests/tests/fixtures/status.go:35
  [FAIL] Operator Upgrades upgrading a cluster with a crash looping sidecar process [It] Upgrade from 6.3.25 to 7.1.27 with crash-looping sidecar
  /codebuild/output/src833382869/src/github.com/FoundationDB/fdb-kubernetes-operator/fdb-kubernetes-tests/tests/fixtures/status.go:35
  [FAIL] Operator Upgrades upgrading a cluster and one coordinator is not restarted [It] Upgrade from 6.3.25 to 7.1.27 with one coordinator not being restarted
  /codebuild/output/src833382869/src/github.com/FoundationDB/fdb-kubernetes-operator/fdb-kubernetes-tests/tests/test_operator_upgrades/operator_upgrades_test.go:443
Can you explain a bit how to find fdb-kubernetes-tests? I've tried https://github.com/FoundationDB/fdb-kubernetes-tests but no luck. Thanks in advance.

Those tests are currently not available as open source, we are working on that.

**Comment by johscheuer** [comment]
Thanks!

**Pull request merged**